builtin/plugins/coin/coin_test.go

```diff
@@ -34,6 +35,105 @@ func (m *mockLoomCoinGateway) DummyMethod(ctx contractpb.Context, req *MintToGat
 	return nil
 }
 
+func TestLoadPolicy(t *testing.T) {
+	contract := &Coin{}
+	pctx := plugin.CreateFakeContext(addr1, addr1)
+	pctx.SetFeature(loomchain.CoinPolicyFeature, true)
+	ctx := contractpb.WrapPluginContext(pctx)
+	//Valid Policy
+	err := contract.Init(ctx, &InitRequest{
+		Accounts: []*InitialAccount{
+			&InitialAccount{
+				Owner:   addr1.MarshalPB(),
+				Balance: uint64(31),
+			},
+		},
+		Policy: &Policy{
+			DeflationFactorDenominator: 5,
+			DeflationFactorNumerator:   1,
+			MintingAccount:             addr1.MarshalPB(),
+		},
+		BaseMintingAmount: 50,
```

Comment:
I thought `BaseMinitingAccount` is inside Policy data structure?

Reply:
Base Minting Amount should be of type uint64 when loaded from genesis.
So in init request it should be outside policy data structure.
In policy it is of type BigUInt, so at the time of policy save in context, it is converted to BigUInt and saved.

Reply:
No, we need to get `BaseMintingAmount` inside `Policy` data structure. It will be inconsistent to have `BaseMintingAmount` outside, since it is logically part of `Policy`.

Plus, BaseMinting amount probably wont become that large that it wont fit uint64. 

So, let's put `BaseMintingAmount` in `Policy` and make it uint64.

Also, no need to check `< 0` for uint64, only `== 0` should suffice.

Reply:
Policy.BaseMintingAMount should be of format - BIgUint.
Overtime, this BaseMintingAMount will aggregate and will become equal to total supply which should be of format - BigUInt 

This is the current - Policy data structure -
message Policy {
    uint64 deflation_factor_numerator = 1 [(gogoproto.customname)="DeflationFactorNumerator"];
    uint64 deflation_factor_denominator = 2 [(gogoproto.customname)="DeflationFactorDenominator"];
    BigUInt base_minting_amount = 3 [(gogoproto.customname)="BaseMintingAmount"];
    Address minting_account = 4 [(gogoproto.customname)="MintingAccount"];
} 

All these calculations should be of format BigUInt- 
policyNumerator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorNumerator))
policyDenominator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorDenominator))
blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
depreciation.Mul(policyNumerator, blockHeight)
depreciation.Div(depreciation, policyDenominator)
amount := policy.BaseMintingAmount.Value
amount.Sub(&amount, depreciation)

If I use uint64 for Policy.BaseMintingAmount, I will have to convert to BigUInt in mint function.
``` 
//Conversion to BigUint required here - 
amount := policy.BaseMintingAmount.Value
amount.Sub(&amount, depreciation)
```

Solution 1 -
1)Keep BaseMintingAmount as uint64 in init request to facilitate data load.
2)Keep BaseMintingAmount as as BigUint in policy.

Solution 2 - 
Keep BaseMintingAmount as unit64 in policy and do conversion to BigUint in Mint Function.

Reply:
Which Solution will you prefer.

Reply:
We will go with solution 2. Keep in mind that while converting do not multiply by 10^18.